Open_Source/OpenSource꞉  6.4 Process to submit a patch to linux kernel.md

```diff
@@ -0,0 +1,226 @@
+## Submitting a patch in Linux Kernel
+
+### Linux Kernel
+
+The Linux Kernel is the main part of a Linux Operating System connecting the hardware and software parts of a computer system. 
+
+The main functions of the kernel are as follows:
+
+<ol>
+    <li><b><i>Memory Management</i></b>: It manages memory being used by different processes.</li>
+    <li><b><i>Process Management</i></b>:Determines the time to be alloted for each process by the Central Processing Unit(CPU)</li>
+    <li><b><i>Device Drivers</i></b>:Connects between hardware and processes</li>
+    <li><b><i>System Security and Calls</i></b>:Receive requests for services from processes</li>
+</ol>
+
+<img src="https://media.geeksforgeeks.org/wp-content/uploads/Linux_Subsystem.png">
+
+### What is a patch?
+
+A patch is a small text document containing a delta of changes between two different versions of a source tree. Patches are created with the 'diff' program.
+
+## Submitting First Patch
+
+### Setting up gmail
+
+Gmail --> Settings --> Forwarding POP/IMAP --> Enable IMAP --> Save Changes
+
+Goto Bottom of page --> Configuration instructions --> Copy the outgoing mail server information to paste it into the .esmtprc file in the next section
+
+### esmtp file configuration 
+
+<b>Announcement</b>: If you already have another mail transfer agent (MTA) installed, installing esmtp is not required. Instead, change the .muttrc file "sendmail" line to be the path to your MTA. Mutt uses ssmtp by default, so if your MTA is ssmtp, you can leave that line out entirely.
+
+First create a .esmtprc file with the required permissions:
+
+Code<br><br>
+
+```
+touch ~/.esmtprc<br>
+chmod g-rwx ~/.esmtprc<br>
+chmod o-rwx ~/.esmtprc```
+
+Editing the .esmtprc file in the home directory:
+
+Code<br><br>
+```
+identity "my.email@gmail.com"<br>
+hostname smtp.gmail.com:587<br>
+username "my.email@gmail.com"<br>
+password "Password"<br>
+starttls required```
```

Comment:
triple backticks should be in the next line , otherwise the console window is not being rendered for this markdown